src/dependency/Dependency/ServiceProviderExtensions/Extensions.GetOrThrow.cs

```diff
@@ -0,0 +1,32 @@
+#nullable enable
+
+using System;
+
+namespace PrimeFuncPack
+{
+    partial class ServiceProviderExtensions
+    {
+        public static T GetServiceOrThrow<T>(
+            this IServiceProvider serviceProvider)
+            where T : notnull
+            =>
+            InternalGetServiceOrThrow<T>(
+                serviceProvider ?? throw new ArgumentNullException(nameof(serviceProvider)));
+
+        private static T InternalGetServiceOrThrow<T>(
+            IServiceProvider serviceProvider)
+            where T : notnull
+            =>
+            serviceProvider.GetService(typeof(T)) switch
```

Comment:
Let's use this method extension:
[https://docs.microsoft.com/en-us/dotnet/api/microsoft.extensions.dependencyinjection.serviceproviderserviceextensions.getservice](url)

`serviceProvider.GetService<T>()`

It seems it should help to avoid processing the third case of unexpected type